borg/xattr.py

```diff
@@ -46,6 +46,14 @@ def get_all(path, follow_symlinks=True):
         logger.error(msg)
         raise Exception(msg)
 
+# If we are running with fakeroot on Linux, then use the xattr functions of fakeroot. This is needed by
+# the 'test_extract_capabilities' test, but also allows xattrs to work with fakeroot on Linux in normal use.
+# TODO: Check whether fakeroot supports xattrs on all platforms supported below.
+# TODO: If that's the case then we can make Borg fakeroot-xattr-compatible on these as well.
+LD_PRELOAD = os.environ.get('LD_PRELOAD', '')
+if sys.platform.startswith('linux') and 'fakeroot' in LD_PRELOAD:
+    libc_name = LD_PRELOAD
```

Comment:
I checked the xattr tests for coverage, all functions are hit. There are some mentions in 2014 changelog entries that they fixed xattrs. I also checked fakeroot-ng, which does *not* support xattr's at all.

I'll add some checks for this.